examples/purificationMBQC/electron_nuclear_spin_register.jl

```diff
@@ -0,0 +1,235 @@
+using ResumableFunctions
+using ConcurrentSim
+using Revise
+using Graphs
+using QuantumSavory
+using QuantumSavory.ProtocolZoo
+import QuantumSavory: Tag
+
+include("../graphstate/graph_preparer.jl")
+
+using Logging
+global_logger(ConsoleLogger(stderr, Logging.Debug))
+
+
+# let's just assume that the overall xor of the measurement results is all we need (maybe wrong) to decide if purification succeeded
+
+@kwdef struct FusionMeasurement
+    node::Int
+    measurement::Int
+end
+Base.show(io::IO, tag::FusionMeasurement) = print(io, "Measurement for register $(tag.node) is $(tag.measurement).")
+Tag(tag::FusionMeasurement) = Tag(FusionMeasurement, tag.node, tag.measurement)
+
+@kwdef struct MBQCMeasurement
+    node::Int
+    measurement::Int
+end
+Base.show(io::IO, tag::MBQCMeasurement) = print(io, "Measurement for register $(tag.node) is $(tag.measurement).")
+Tag(tag::MBQCMeasurement) = Tag(MBQCMeasurement, tag.node, tag.measurement)
+
+@kwdef struct XORMeasurements
+    node::Int
+    xor_result::Int
+end
+Base.show(io::IO, tag::XORMeasurements) = print(io, "XOR measurement for register $(tag.node) is $(tag.xor_result).")
+Tag(tag::XORMeasurements) = Tag(XORMeasurements, tag.node, tag.xor_result)
+
+@kwdef struct PurifiedEntalgementCounterpart
+    remote_node::Int
+    remote_slot::Int
+end
+Base.show(io::IO, tag::PurifiedEntalgementCounterpart) = print(io, "Entangled to $(tag.remote_node).$(tag.remote_slot)")
+Tag(tag::PurifiedEntalgementCounterpart) = Tag(PurifiedEntalgementCounterpart, tag.remote_node, tag.remote_slot)
+
+
+# TODO: hardcoded for now
+function graph_generator(initial_entanglements_nodes, purified_nodes)
+    g = Graph(4)
+    for ij in [(1,3), (2,3), (3,4)]
+        add_edge!(g, ij...)
+    end
+    return g
+end
+
+# TODO: right now, it just measures all but purified nodes
+@resumable function measure(sim, net, side, purified_nodes, storage_slot)
```

Comment:
let's avoid assuming that this will be the only protocol running (e.g. the way you do index arithmetics with n depends on the assumption you have only two groups of nodes, only dedicated to running this protocol)

maybe;

`function measure(sim, net, local_storage_nodes_idx, storage_slot, remote_chief_node_idx)`